src/sentry/static/sentry/app/views/projects/createProjectButton.jsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Modal from 'react-bootstrap/lib/Modal';
+
+import {t} from 'app/locale';
+import {FormState} from 'app/components/forms';
+import {ValidationIssues} from 'app/components/validationIssues';
+import IndicatorStore from 'app/stores/indicatorStore';
+
+class CreateProjectButton extends React.Component {
+  // TODO-simple: lab is hardcoded as an org
+  constructor(props) {
+    super(props);
+    this.state = {
+      isModalOpen: false,
+      value: null,
+      selectedFile: null,
+      loaded: 0,
+      validationIssues: [],
+      errorMsg: null,
+    };
+
+    this.onToggle = this.onToggle.bind(this);
+  }
+
+  onToggle() {
+    if (this.props.disabled) {
+      return;
+    }
+
+    this.setState({
+      isModalOpen: !this.state.isModalOpen,
+      state: FormState.READY,
+      errorMsg: null,
+      validationIssues: [],
+    });
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <a
+          title={this.props.tooltip || this.props.buttonTitle}
+          className="btn btn-sm btn-default"
+          disabled={this.props.disabled}
+          onClick={this.onToggle}
```

Comment:
I'm expecting to see some create project form to show here? How is it that a create project button is linked to onToggle event?

Reply:
Right! This is copied from e.g. the `AssignToWorkflowButton`. I think the naming is confusing, a this is both a view for the button and the modal window. Should rename to `Modal` or something (copied this naming convention blindly from sentry).

The onToggle function toggles between showing and not showing, so when clicking the button it leads to a toggle.